utils/providers.py

```diff
@@ -496,6 +481,88 @@ def destroy_vm(provider_mgmt, vm_name):
         logger.error('%s destroying VM %s (%s)', type(e).__name__, vm_name, e.message)
 
 
+def get_credentials_from_config(credential_config_name):
+    creds = conf.credentials[credential_config_name]
+    return BaseProvider.Credential(principal=creds['username'],
+                               secret=creds['password'])
+
+
+def get_crud(provider_config_name):
+    """
+    Creates a Provider object given a yaml entry in cfme_data.
+
+    Usage:
+        get_crud('ec2east')
+
+    Returns: A Provider object that has methods that operate on CFME
+    """
+
+    prov_config = conf.cfme_data.get('management_systems', {})[provider_config_name]
+    credentials = get_credentials_from_config(prov_config['credentials'])
+    prov_type = prov_config.get('type')
+
+    if prov_type != 'ec2':
+        if prov_config.get('discovery_range', None):
+            start_ip = prov_config['discovery_range']['start']
+            end_ip = prov_config['discovery_range']['end']
+        else:
+            start_ip = prov_config['ipaddress']
+            end_ip = prov_config['ipaddress']
+
+    if prov_type == 'ec2':
+        return EC2Provider(name=prov_config['name'],
+            region=prov_config['region'],
+            credentials={'default': credentials},
+            zone=prov_config['server_zone'],
+            key=provider_config_name)
+    elif prov_type == 'openstack':
+        return OpenStackProvider(name=prov_config['name'],
+            hostname=prov_config['hostname'],
+            ip_address=prov_config['ipaddress'],
+            api_port=prov_config['port'],
+            credentials={'default': credentials},
+            zone=prov_config['server_zone'],
+            key=provider_config_name)
+    elif prov_type == 'virtualcenter':
+        return VMwareProvider(name=prov_config['name'],
+            hostname=prov_config['hostname'],
+            ip_address=prov_config['ipaddress'],
+            credentials={'default': credentials},
+            zone=prov_config['server_zone'],
+            key=provider_config_name,
+            start_ip=start_ip,
+            end_ip=end_ip)
+    elif prov_type == 'scvmm':
```

Comment:
SCVMM used to have special credential handling for the "domain" portion. Now it doesn't need it as it's handled later.